codec-http2/src/main/java/io/netty/handler/codec/h2new/DefaultChannelFlowControlledBytesDistributor.java

```diff
@@ -0,0 +1,231 @@
+        // https://httpwg.org/specs/rfc7540.html#WINDOW_UPDATE
+
+        Integer leftover = handleIOClosed(streamId, localAcceptors);
+        if (leftover != null) {
+            handleLeftOverBytesLocal(streamId, leftover);
+        }
+    }
+
+    private Integer handleIOClosed(int streamId, IntObjectMap<Object> acceptor) {
+        final Object removed = acceptor.remove(streamId);
+        if (removed == null) {
+            return null;
+        }
+        final int leftover;
+        if (removed instanceof DistributionAcceptor) {
+            leftover = ((DistributionAcceptor) removed).dispose();
+        } else {
+            assert removed instanceof Integer;
+            leftover = (int) removed;
+        }
+        return leftover;
+    }
+
+    void newLocallyInitiatedStream(int streamId) {
+        onNewStream(streamId, null);
+    }
+
+    private void handleReprioritization(Http2PriorityFrame http2Frame) {
+    }
+
+    private void handleLeftOverBytesRemote(int streamId, int dispose) {
+    }
+
+    private void handleLeftOverBytesLocal(int streamId, int dispose) {
+    }
+
+    private void onNewStream(int streamId, Http2HeadersFrame headersFrame) {
+        initAcceptorForNewStream(streamId, headersFrame, this::forNewStreamWrite, remoteAcceptors);
+        initAcceptorForNewStream(streamId, headersFrame, this::forNewStreamRead, localAcceptors);
+    }
+
+    private void initAcceptorForNewStream(int streamId, Http2HeadersFrame headersFrame,
+                                          HeadersToIntFunction newStreamProvider,
+                                          IntObjectMap<Object> acceptrs) {
+        acceptrs.compute(streamId, (id, existing) -> {
+            if (existing == null) {
+                return newStreamProvider.apply(headersFrame);
+            }
+            if (existing instanceof DistributionAcceptor) {
+                DistributionAcceptor acceptor = (DistributionAcceptor) existing;
+                if (acceptor instanceof EarlyAcceptor) {
+                    acceptor.accumulate(newStreamProvider.apply(headersFrame));
+                }
+                return existing;
+            }
+            assert existing instanceof Integer;
+            return existing;
+        });
+    }
+
+    private int forNewStreamWrite(Http2HeadersFrame headersFrame) {
+        // check priority, weights and distribute
+        return 65_535; // https://httpwg.org/specs/rfc7540.html#InitialWindowSize
+    }
+
+    private int forNewStreamRead(Http2HeadersFrame headersFrame) {
+        // check priority, weights and distribute
+        return 65_535; // https://httpwg.org/specs/rfc7540.html#InitialWindowSize
+    }
+
+    private DistributionAcceptor replace(int streamId, IntObjectMap<Object> acceptors, DistributionAcceptor acceptor) {
+        if (!channel.executor().inEventLoop()) {
+            throw new IllegalArgumentException("Invalid caller, not on eventloop: " + channel.executor());
```

Comment:
How about using `IllegalAccessException`?

Reply:
`IllegalAccessException` is reflection specific. `IllegalStateException` would be better.